mms/data_store/data_store.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""
+Data Store for use in back-end batching of requests
+"""
+
+from mms.data_store.codecs import JsonCodec, ImageCodec
+from mms.data_store.hashmap import RedisHashMap
+from mms.data_store.queue import RedisQueue
+
+
+class DataStore(object):
+    """
+    DataStore abstracts the queue, hashmap, and associated codecs
+    """
+
+    def __init__(self, name_prefix, config):
+        self.hashmap = RedisHashMap(name_prefix, config)
+        self.queue = RedisQueue(name_prefix, config)
+
+    def pop_batch(self, model_name, length, data_type):
+        codec = self._get_codec(data_type)
```

Comment:
Maybe a good idea to check `codec` == none here and assert or just assert/raise exception in get_codec.